compiler/ballerina-lang/src/main/resources/grammar/BallerinaParser.g4

```diff
@@ -121,8 +121,9 @@ constantDefinition
     ;
 
 globalVariableDefinition
-    :   PUBLIC? LISTENER? typeName Identifier (ASSIGN expression)? SEMICOLON
-    |   PUBLIC? FINAL? (typeName | VAR) Identifier ASSIGN expression SEMICOLON
+    :   PUBLIC? LISTENER typeName Identifier (ASSIGN expression)? SEMICOLON
+    |   FINAL (typeName | VAR) Identifier ASSIGN expression SEMICOLON
+    |   (typeName | VAR) Identifier (ASSIGN expression)? SEMICOLON
```

Comment:
If I'm not mistaken. in this rule (3rd) if `var` is used, the expression can be optional. But it should be mandatory.

Reply:
Thanks, missed this. Fixed with updates to the grammar and added tests.